runtime/trappist/src/impls.rs

```diff
@@ -111,6 +112,34 @@ impl xcm_primitives::PauseXcmExecution for XcmExecutionManager {
 		XcmpQueue::resume_xcm_execution(RuntimeOrigin::root())
 	}
 }
+
+pub trait WeightCoefficientCalc<Balance> {
+	fn saturating_eval(&self, result: Balance, x: Balance) -> Balance;
+}
+
+impl<Balance> WeightCoefficientCalc<Balance> for WeightToFeeCoefficient<Balance>
+where
+	Balance: BaseArithmetic + From<u32> + Copy + Unsigned + SaturatedConversion,
+{
+	fn saturating_eval(&self, mut result: Balance, x: Balance) -> Balance {
+		let power = x.saturating_pow(self.degree.into());
+
+		let frac = self.coeff_frac * power; // Overflow safe.
```

Comment:
I think you can be more specific here and say that it's "Overflow safe" because `coeff_frac` is strictly less than 1.